src/groupeddataframe/grouping.jl

```diff
@@ -202,23 +238,185 @@ df = DataFrame(a = repeat([1, 2, 3, 4], outer=[2]),
                b = repeat([2, 1], outer=[4]),
                c = randn(8))
 gd = groupby(df, :a)
-combine(map(d -> mean(skipmissing(d[:c])), gd))
+combine(d -> sum(skipmissing(d[:c])), gd)
 ```
 
+### See also
+
+[`by(f, df, cols)`](@ref) is a shorthand for `combine(f, groupby(df, cols))`.
+
+[`map`](@ref): `combine(f, groupby(df, cols))` is a more efficient equivalent
+of `combine(map(f, groupby(df, cols)))`.
+
 """
-function combine(ga::GroupApplied)
-    gd, vals = ga.gd, ga.vals
-    valscat = _vcat(vals)
-    idx = Vector{Int}(undef, size(valscat, 1))
-    j = 0
-    @inbounds for (start, val) in zip(gd.starts, vals)
-        n = size(val, 1)
-        idx[j .+ (1:n)] .= gd.idx[start]
-        j += n
+function combine(f::Function, gd::GroupedDataFrame)
+    if length(gd) > 0
+        idx, valscat = _combine(wrap(f(gd[1])), f, gd)
+        return hcat!(gd.parent[idx, gd.cols], valscat, makeunique=true)
```

Comment:
I had to put `makeunique=true` to get some of the tests passing, in particular `by(identity, ...)`. I guess that's OK for now until we tackle #1460 more directly. Indeed, I'm afraid it would be too annoying not to allow people to return the same data frame as the input.

Reply:
Agreed, at least we are specific now, so it is easier to track cases like this.